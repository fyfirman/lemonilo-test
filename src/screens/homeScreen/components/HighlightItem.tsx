import React from 'react';
import styled from 'styled-components/native';

const Container = styled.TouchableOpacity`
  width: 150px;
  margin-right: ${({ theme }) => theme.spacing(2)}px;
`;

const Photo = styled.Image`
  border-radius: ${({ theme }) => theme.borderRadius.small}px;
  aspect-ratio: 1;
`;

const Name = styled.Text`
  color: ${({ theme }) => theme.palette.text.primary};
`;

const Price = styled.Text`
  color: ${({ theme }) => theme.palette.text.secondary};
`;

export interface HighlightItemData {
  name: string;
  price: number;
  rating?: number;
  imageUri: string;
}

interface HighlightItemProps extends HighlightItemData {
  onPress: () => any;
}

const HighlightItem = (props: HighlightItemProps) => {
  const { name, price, imageUri, onPress } = props;

  return (
    <Container onPress={onPress}>
      <Photo source={{ uri: imageUri }} resizeMode="cover" />
      <Name>{name}</Name>
      <Price>{`Rp ${price}`}</Price>
    </Container>
  );
};

export default HighlightItem;
