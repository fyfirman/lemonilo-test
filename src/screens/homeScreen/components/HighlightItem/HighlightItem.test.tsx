import React from 'react';
import { renderWithTheme } from '../../../../__tests__/helpers';
import HighlightItem from './HighlightItem';
import 'jest-styled-components';

test('should match with snapshot', () => {
  const data = {
    name: 'Joy N Fun Festival City Link',
    imageUri:
      'https://ik.imagekit.io/tvlk/xpe-asset/AyJ40ZAo1DOyPyKLZ9c3RGQHTP2oT4ZXW+QmPVVkFQiXFSv42UaHGzSmaSzQ8DO5QIbWPZuF+VkYVRk6gh-Vg4ECbfuQRQ4pHjWJ5Rmbtkk=/2001871272969/Joy%2520N%2520Fun%2520Festival%2520City%2520Link-61055bfc-c44e-4588-a66a-b99d86b7ba4f.jpeg?_src=imagekit&tr=c-at_max,h-1280,q-60,w-720',
    price: 48500,
  };

  const tree = renderWithTheme(
    <HighlightItem
      imageUri={data.imageUri}
      name={data.name}
      onPress={jest.fn}
      price={data.price}
    />,
  );

  expect(tree).toMatchSnapshot();
});
