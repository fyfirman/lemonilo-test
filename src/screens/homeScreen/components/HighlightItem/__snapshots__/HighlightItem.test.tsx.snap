// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 16,
      "opacity": 1,
      "width": 150,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "https://ik.imagekit.io/tvlk/xpe-asset/AyJ40ZAo1DOyPyKLZ9c3RGQHTP2oT4ZXW+QmPVVkFQiXFSv42UaHGzSmaSzQ8DO5QIbWPZuF+VkYVRk6gh-Vg4ECbfuQRQ4pHjWJ5Rmbtkk=/2001871272969/Joy%2520N%2520Fun%2520Festival%2520City%2520Link-61055bfc-c44e-4588-a66a-b99d86b7ba4f.jpeg?_src=imagekit&tr=c-at_max,h-1280,q-60,w-720",
      }
    }
    style={
      Array [
        Object {
          "aspectRatio": 1,
          "borderRadius": 4,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "rgba(0,0,0,0.8)",
        },
      ]
    }
  >
    Joy N Fun Festival City Link
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(0,0,0,0.5)",
        },
      ]
    }
  >
    Rp 48.500
  </Text>
</View>
`;
